zulipterminal/ui_tools/boxes.py

```diff
@@ -119,27 +122,32 @@ def autocomplete_mentions(self, text: str, state: int,
                           for user in users_list
                           if match_user(user, text[len(prefix_string):])]
         combined_typeahead = group_typeahead + user_typeahead
+        combined_typeahead = combined_typeahead[:num_suggestions]
```

Comment:
While this reuses a variable name, my preference would be to give it a different name (`limited_typeahead`?) or to simply place the `:num_suggestions` into the try clause.